### ISSUE TITLE
dial tcp: lookup mysqlclustertest.blob.core.windows.net on 127.0.0.53:53: no such host

### ISSUE THREAD

**Issue by kangvai** [rclone/rclone#7083]
#### What is the problem you are having with rclone?

After i config rclone, use `rclone lsd remote:`, it prompt.
![image](https://github.com/rclone/rclone/assets/59158970/b9737056-17da-4134-99c6-ef7416f01978)

I found in `azureblob.go` line 65, the `storageDefaultBaseURL = "blob.core.windows.net"`
but in china, azure Operated by 21vianet and the blob service url usually like this `*.blob.core.chinacloudapi.cn`
Can support for this suffix be added ？Looking forward to your reply! 

#### What is your rclone version (output from `rclone version`)
```
root@test-node-2:~# rclone version
rclone v1.62.2
- os/version: ubuntu 22.04 (64 bit)
- os/kernel: 5.15.0-1040-azure (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.20.2
- go/linking: static
- go/tags: none
```

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)
- os/version: ubuntu 22.04 (64 bit)

#### Which cloud storage system are you using? (e.g. Google Drive)
- azureblob

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)
```
rclone lsd remote:
```

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)
```
root@test-node-2:~# rclone -vvv lsd remote:
2023/06/26 03:46:46 DEBUG : rclone: Version "v1.62.2" starting with parameters ["rclone" "-vvv" "lsd" "remote:"]
2023/06/26 03:46:46 DEBUG : Creating backend with remote "remote:"
2023/06/26 03:46:46 DEBUG : Using config file from "/root/.config/rclone/rclone.conf"
2023/06/26 03:46:54 ERROR : : error listing: dial tcp: lookup mysqlclustertest.blob.core.windows.net on 127.0.0.53:53: no such host
2023/06/26 03:46:54 DEBUG : 2 go routines active
2023/06/26 03:46:54 Failed to lsd with 2 errors: last error was: dial tcp: lookup mysqlclustertest.blob.core.windows.net on 127.0.0.53:53: no such host
```

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by ncw** [comment]
When configuring rclone did you set the `region` properly?

```
Option region.
Choose national cloud region for OneDrive.
Choose a number from below, or type in your own string value.
Press Enter for the default (global).
 1 / Microsoft Cloud Global
   \ (global)
 2 / Microsoft Cloud for US Government
   \ (us)
 3 / Microsoft Cloud Germany
   \ (de)
 4 / Azure and Office 365 operated by Vnet Group in China
   \ (cn)
region> 

```

You want `region = cn` in the config file I think and it should work properly.

**Comment by kapitainsky** [comment]
> When configuring rclone did you set the `region` properly?

It is possible for onedrive but not for `Microsoft Azure Blob Storage` remote

> I found in `azureblob.go` line 65, the `storageDefaultBaseURL = "blob.core.windows.net"`

@kangvai could you build `rclone` from source changing this line to `storageDefaultBaseURL = "blob.core.chinacloudapi.cn"` and see if it works in China? .  If you need instructions ask.

**Comment by ncw** [comment]
Sorry @kangvai didn't read your question properly!

You can set the endpoint with `endpoint = https://YOURACCOUNT.blob.core.chinacloudapi.cn` - can you see if that works?

If that works we should probably add a `region`. Are there any microsoft docs about different regions do you know?

**Comment by kangvai** [comment]
@kapitainsky Thanks for you answer ! Yeah After I build `rclone` from source changing this line to `storageDefaultBaseURL = "blob.core.chinacloudapi.cn"`, it does work in China.

**Comment by kangvai** [comment]
@ncw Thank you, I will test this later. About endpoints different between Azure China and Azure Global you can see the link blew. For other regions, sorry i don't know.
[https://docs.azure.cn/zh-cn/articles/guidance/developerdifferences#check-endpoints-in-azure](url)

**Comment by animosity22** [comment]
Looks like something might be geoblocked or something I'd imagine.

If you are working, do you need anything else as this looks answered?

**Comment by kangvai** [comment]
Yes, I don't need anything else. Thanks, but I hope can support azure in China by add some features. like this

> When configuring rclone did you set the `region` properly?
> 
> ```
> Option region.
> Choose national cloud region for OneDrive.
> Choose a number from below, or type in your own string value.
> Press Enter for the default (global).
>  1 / Microsoft Cloud Global
>    \ (global)
>  2 / Microsoft Cloud for US Government
>    \ (us)
>  3 / Microsoft Cloud Germany
>    \ (de)
>  4 / Azure and Office 365 operated by Vnet Group in China
>    \ (cn)
> region> 
> ```
> 
> You want `region = cn` in the config file I think and it should work properly.

**Comment by ncw** [comment]
The first step I guess would be you telling us if it worked or not @kangvai 

If it did then we can put some docs in at minimum, but a region flag would be a good idea

**Comment by kapitainsky** [comment]
> @kapitainsky Thanks for you answer ! Yeah After I build `rclone` from source changing this line to `storageDefaultBaseURL = "blob.core.chinacloudapi.cn"`, it does work in China.

It was tested by replacing storageDefaultBaseURL in source code and confirmed working.

**Comment by ncw** [comment]
@kangvai can you check if this works without changing the source code please?

> You can set the endpoint with `endpoint = https://YOURACCOUNT.blob.core.chinacloudapi.cn` - can you see if that works?

**Comment by HaxiSnake** [comment]
> @kangvai can you check if this works without changing the source code please?
> 
> > You can set the endpoint with `endpoint = https://YOURACCOUNT.blob.core.chinacloudapi.cn` - can you see if that works?

@ncw 

I've encountered the same issue in version v1.68.1, but it got resolved after I changed the endpoint like this. However, in version v1.58, using the endpoint blob.core.chinacloudapi.cn worked without any problems. FYI
